File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/entities/dtos/SplitDto.java
package com.decrux.pocketr.api.entities.dtos;

import java.util.UUID;

public record SplitDto(
    UUID id,
    UUID accountId,
    String accountName,
    String accountType,
    String side,
    long amountMinor,
    long effectMinor,
    UUID categoryTagId,
    String categoryTagName
) {
    public UUID getId() {
        return id;
    }

    public UUID getAccountId() {
        return accountId;
    }

    public String getAccountName() {
        return accountName;
    }

    public String getAccountType() {
        return accountType;
    }

    public String getSide() {
        return side;
    }

    public long getAmountMinor() {
        return amountMinor;
    }

    public long getEffectMinor() {
        return effectMinor;
    }

    public UUID getCategoryTagId() {
        return categoryTagId;
    }

    public String getCategoryTagName() {
        return categoryTagName;
    }
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr_api/entities/dtos/SplitDto.kt
package com.decrux.pocketr_api.entities.dtos

import java.util.UUID

data class SplitDto(
    val id: UUID,
    val accountId: UUID,
    val accountName: String,
    val accountType: String,
    val side: String,
    val amountMinor: Long,
    val effectMinor: Long,
    val categoryTagId: UUID?,
    val categoryTagName: String?,
)
